hilti/toolchain/include/ast/declaration.h

```diff
@@ -45,6 +47,54 @@ namespace linkage {
 constexpr auto from_string(const std::string_view& s) { return util::enum_::from_string<Linkage>(s, detail::linkages); }
 } // namespace linkage
 
+/** Represents a declaration's documentation string. */
+class DocString {
+public:
+    /**
+     * Returns all lines of summary text added so far. The returned lines will
+     * have their comment prefixes stripped.
+     */
+    const auto& summary() const { return _summary; }
+
+    /**
+     * Returns all lines of documentation text added so far. The returned lines
+     * will have their comment prefixes stripped.
+     */
+    const auto& text() const { return _text; }
+
+    /**
+     * Appends a line of summary text to the documentation.
+     *
+     * @param line line to add, with optional comment prefix (which will be removed)
+     */
+    void addSummary(const std::string& line) { _summary.push_back(normalize(line)); }
+
+    /**
+     * Appends a line of documentation text to the documentation.
+     *
+     * @param line line to add, with optional comment prefix (which will be removed)
+     */
+    void addText(const std::string& line) { _text.push_back(normalize(line)); }
+
+    /** Empties out all content. */
+    void clear();
+
+    /**
+     * Renders the comment back into a multi-line string. This is primarily for debugging.
+     */
+    void render(std::ostream& out) const;
+
+    /** Returns true if any summary or documentation text has been added. */
+    operator bool() const { return ! (_summary.empty() || _text.empty()); }
```

Comment:
Let's disable conversion of this type to e.g., `int`.

```suggestion
    explicit operator bool() const { return ! (_summary.empty() || _text.empty()); }
```